drivers/volume/portworx/portworx.go

```diff
@@ -428,6 +476,29 @@ func (p *portworx) SnapshotCreate(
 		return nil, getErrorSnapshotConditions(err), err
 	}
 
+	if snap.Metadata.Annotations != nil {
+		preSnapRule, present := snap.Metadata.Annotations[preSnapRuleAnnotationKey]
+		if present && len(preSnapRule) > 0 {
+			logrus.Infof("Running pre-snap rule: %s for snapshot: [%s] %s", preSnapRule, snap.Metadata.Namespace, snap.Metadata.Name)
+			taskID, _ := uuid.New()
+
+			podsWithBackgroundCmds, err := p.executeRuleForSnap(snap, preSnapRule, taskID.String())
```

Comment:
This can't be waiting for cloudsnap to finish which can take a long time. For cloudsnap the pre-snap command can be stopped after cloudsnap has been triggered

Reply:
done